Source/WebKit/WebProcess/WebPage/ios/WebPageIOS.mm

```diff
@@ -1242,8 +1242,38 @@ static void dispatchSyntheticMouseMove(LocalFrame& mainFrame, const WebCore::Flo
 
 void WebPage::potentialTapAtPosition(WebKit::TapIdentifier requestID, const WebCore::FloatPoint& position, bool shouldRequestMagnificationInformation)
 {
-    if (RefPtr localMainFrame = dynamicDowncast<LocalFrame>(m_page->mainFrame()))
+    RefPtr localMainFrame = dynamicDowncast<LocalFrame>(m_page->mainFrame());
+    if (localMainFrame)
         m_potentialTapNode = localMainFrame->nodeRespondingToClickEvents(position, m_potentialTapLocation, m_potentialTapSecurityOrigin.get());
+
+    auto lastTouchLocation = std::exchange(m_lastTouchLocationBeforeTap, { });
+    bool ignorePotentialTap = [&] {
+        if (!m_potentialTapNode)
+            return false;
+
+        if (!localMainFrame)
+            return false;
+
+        if (!lastTouchLocation)
+            return false;
+
+        static constexpr auto maxAllowedMovement = 200;
+        if ((position - *lastTouchLocation).diagonalLengthSquared() <= std::pow(maxAllowedMovement, 2))
```

Comment:
TIL `std::pow()` is not `constexpr`... I would've just suggested this to prevent the runtime exponentiation cost:

```cpp
static constexpr auto maxAllowedDiagonalMovement = std::pow(200, 2); // or 200 * 200
```

Reply:
C++26 maybe 🤷 https://wg21.link/p1383r2